lib/logstash/inputs/http.rb

```diff
@@ -57,28 +40,61 @@ class LogStash::Inputs::Http < LogStash::Inputs::Base
   # The TCP port to bind to
   config :port, :validate => :number, :default => 8080
 
-  # Maximum number of threads to use
-  config :threads, :validate => :number, :default => 4
+  # If a connection request arrives when the queue is full, the connection is refused
+  config :backlog, :validate => :number, :default => 128
 
   # Username for basic authorization
   config :user, :validate => :string, :required => false
 
   # Password for basic authorization
   config :password, :validate => :password, :required => false
 
-  # SSL Configurations
-  #
-  # Enable SSL
+  # Events are by default sent in plain text. You can
+  # enable encryption by setting `ssl` to true and configuring
+  # the `ssl_certificate` and `ssl_key` options.
   config :ssl, :validate => :boolean, :default => false
 
-  # The JKS keystore to validate the client's certificates
-  config :keystore, :validate => :path
+  # SSL certificate to use.
```

Comment:
These SSL settings are bit confusing.  I would suggest to present configuration in the form of keystore and truststore's.  Presumption that both are PKCS#12 backed java keystores, and _may_ be the same file. 

Keystore holds the Pb key (certs) it presents to clients for verification and the Pr key it uses for TLS. 
Truststore holds the Pb keys (certs) for the all the entire chain of trust for mutual (peer) TLS such as the self-signed, intermediate, CA certs, etc. 

specifically : ssl_key -> keystore , ssl_key_passphrase -> keystore_password, ssl_certficate_authorities -> trust_store, (add trust_store_password)

I think we should use this opportunity to default to true for SSL/TLS (and encourage, but not require mutual (peer) TLS).

specifically : ssl -> true (not false) , and ssl_verify_mode -> ssl_peer_verify_mode = [none, verify, force_verify] (with verify the default and warns if client does present a cert)

Reply:
I see now that these are the same settings as beats, but the recommendation that I made aligns us closer to the ES output (which I think is more correct).  This probably requires more discussion on convergence of configuration options.

Reply:
Yeah, I used beats code and settings since it had the most complete set of settings for ssl handling (support for intermediate cas, control over ciphers, tls min/max version setting, etc.

looking at other places in the stack, namely x-pack security, we tend to prefer PEM format over jks.

That said ++ on a broader discussion of common ssl settings